x/upgrade/integration_test.go

```diff
@@ -0,0 +1,57 @@
+package upgrade_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/app"
+	testutil "github.com/celestiaorg/celestia-app/test/util"
+	"github.com/celestiaorg/celestia-app/x/upgrade/types"
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+	tmtypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+func TestUpgradeIntegration(t *testing.T) {
+	app, _ := testutil.SetupTestAppWithGenesisValSet(app.DefaultConsensusParams())
+	ctx := sdk.NewContext(app.CommitMultiStore(), tmtypes.Header{
+		Version: tmversion.Consensus{
+			App: 1,
+		},
+	}, false, tmlog.NewNopLogger())
+	goCtx := sdk.WrapSDKContext(ctx)
+	ctx = sdk.UnwrapSDKContext(goCtx)
+
+	res, err := app.UpgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 0, res.VotingPower)
+
+	validators := app.StakingKeeper.GetAllValidators(ctx)
+	valAddr, err := sdk.ValAddressFromBech32(validators[0].OperatorAddress)
+	require.NoError(t, err)
+
+	_, err = app.UpgradeKeeper.SignalVersion(ctx, &types.MsgSignalVersion{
+		ValidatorAddress: valAddr.String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err = app.UpgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 1, res.VotingPower)
+	require.EqualValues(t, 0, res.ThresholdPower)
```

Comment:
[optional] this threshold power seems weird b/c it makes me think that no votes are needed to trigger an upgrade.
 Is it possible for this test case to have a non-zero threshold power potentially by adding another validator with some voting power?